/*
 * This file was automatically generated by EvoSuite
 * Sat Apr 16 13:48:09 GMT 2022
 */

package org.apache.tika.sax;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.tika.sax.TaggedContentHandler;
import org.apache.tika.sax.TaggedSAXException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.lang.MockException;
import org.junit.runner.RunWith;
import org.xml.sax.ContentHandler;
import org.xml.sax.SAXException;
import org.xml.sax.SAXParseException;
import org.xml.sax.helpers.XMLFilterImpl;
import org.xml.sax.helpers.XMLReaderAdapter;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class TaggedContentHandler_ESTest extends TaggedContentHandler_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      XMLReaderAdapter xMLReaderAdapter0 = new XMLReaderAdapter();
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler(xMLReaderAdapter0);
      SAXParseException sAXParseException0 = new SAXParseException("rS", "rS", "rS", 604, 604);
      TaggedSAXException taggedSAXException0 = new TaggedSAXException(sAXParseException0, taggedContentHandler0);
      try { 
        taggedContentHandler0.throwIfCauseOf(taggedSAXException0);
        fail("Expecting exception: SAXParseException");
      
      } catch(SAXParseException e) {
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler((ContentHandler) null);
      // Undeclared exception!
      try { 
        taggedContentHandler0.handleException((SAXException) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.tika.sax.TaggedSAXException", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      XMLFilterImpl xMLFilterImpl0 = new XMLFilterImpl();
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler(xMLFilterImpl0);
      SAXException sAXException0 = new SAXException();
      TaggedSAXException taggedSAXException0 = new TaggedSAXException(sAXException0, taggedContentHandler0);
      try { 
        taggedContentHandler0.throwIfCauseOf(taggedSAXException0);
        fail("Expecting exception: SAXException");
      
      } catch(SAXException e) {
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      XMLReaderAdapter xMLReaderAdapter0 = new XMLReaderAdapter();
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler(xMLReaderAdapter0);
      SAXParseException sAXParseException0 = new SAXParseException("}`6tN;Rm", "}`6tN;Rm", "}`6tN;Rm", 604, 604);
      Object object0 = new Object();
      TaggedSAXException taggedSAXException0 = new TaggedSAXException(sAXParseException0, object0);
      taggedContentHandler0.throwIfCauseOf(taggedSAXException0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      XMLReaderAdapter xMLReaderAdapter0 = new XMLReaderAdapter();
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler(xMLReaderAdapter0);
      SAXParseException sAXParseException0 = new SAXParseException("}`6tN;Rm", "}`6tN;Rm", "}`6tN;Rm", 604, 604);
      taggedContentHandler0.throwIfCauseOf(sAXParseException0);
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      XMLReaderAdapter xMLReaderAdapter0 = new XMLReaderAdapter();
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler(xMLReaderAdapter0);
      SAXParseException sAXParseException0 = new SAXParseException("}`6tN;Rm", "}`6tN;Rm", "}`6tN;Rm", 604, 604);
      Object object0 = new Object();
      TaggedSAXException taggedSAXException0 = new TaggedSAXException(sAXParseException0, object0);
      boolean boolean0 = taggedContentHandler0.isCauseOf(taggedSAXException0);
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler((ContentHandler) null);
      MockException mockException0 = new MockException("C");
      SAXException sAXException0 = new SAXException(mockException0);
      boolean boolean0 = taggedContentHandler0.isCauseOf(sAXException0);
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler((ContentHandler) null);
      MockException mockException0 = new MockException("C");
      SAXException sAXException0 = new SAXException(mockException0);
      try { 
        taggedContentHandler0.handleException(sAXException0);
        fail("Expecting exception: TaggedSAXException");
      
      } catch(TaggedSAXException e) {
         //
         // C
         //
         verifyException("org.apache.tika.sax.TaggedContentHandler", e);
      }
  }

  @Test(timeout = 4000)
  public void test8()  throws Throwable  {
      XMLFilterImpl xMLFilterImpl0 = new XMLFilterImpl();
      TaggedContentHandler taggedContentHandler0 = new TaggedContentHandler(xMLFilterImpl0);
      SAXException sAXException0 = new SAXException();
      TaggedSAXException taggedSAXException0 = new TaggedSAXException(sAXException0, taggedContentHandler0);
      boolean boolean0 = taggedContentHandler0.isCauseOf(taggedSAXException0);
      assertTrue(boolean0);
  }
}
